src/server/routes/login.js

```diff
@@ -32,19 +31,19 @@ const login = async (
     return response.status(401).send('Invalid credentials')
   }
 
-  let token = await TokenModel.retrieveToken(user)
+  let token = await retrieveToken(user)
 
   if (token === null) {
-    const tokenResponse = await TokenModel.createToken(user)
+    const tokenResponse = await createToken(user)
 
-    if (tokenResponse.token === null) {
+    if (tokenResponse.token === null || tokenResponse.errors.length > 0) {
       return response.status(500).send('An error occurred.')
     }
 
     token = tokenResponse.token
   }
 
-  return response.json({ token })
+  return response.json({ token: token.token })
```

Comment:
`token.token`? Isn’t that gonna cause issues in some cases? Line 43 you set it to `tokenResponse.token` already.

Reply:
Alright this one is a bit confusing, I should improve naming. `tokenResponse.token` refers to a Token model (token field of TokenResponse type), but what we want to return here is only the actual token string, which is the `token` property of the `Token` model. It would probably be clearer using `tokenString` for the property name (thus giving a nicer `token.tokenString` as return value)